addons/stock/models/stock_location.py

```diff
@@ -241,7 +241,14 @@ def _get_putaway_strategy(self, product, quantity=0, package=None, packaging=Non
             for values in quant_data:
                 qty_by_location[values['location_id'][0]] += values['quantity']
 
-        return putaway_rules._get_putaway_location(product, quantity, package, qty_by_location) or self
+        putaway_location = putaway_rules._get_putaway_location(product, quantity, package, qty_by_location) or self
+        # A putaway location might not be found, but a move_line shouldn't target a 'view' location. So search the first valid child.
+        if putaway_location.usage == 'view':
+            putaway_location = self.env['stock.location'].search(
+                [('id', 'child_of', putaway_location.id), ('usage', '=', 'internal'), ('usage', '!=', 'view')],
```

Comment:
`('usage', '=', 'internal'), ('usage', '!=', 'view')` the second part is useless.